src/api/login.ts

```diff
@@ -1,46 +1,35 @@
-import axios from 'axios';
-import toast from 'react-hot-toast';
+import apiClient from '@/api/apiClient';
 
-const api = axios.create({
-  baseURL: import.meta.env.VITE_API_BASE_URL,
-  headers: {
-    'Content-Type': 'application/json',
-  },
-});
+interface LoginRequest {
+  email: string;
+  password: string;
+}
 
-api.interceptors.request.use(
-  (config) => {
-    const userInfo = sessionStorage.getItem('userInfo');
-    if (userInfo) {
-      try {
-        const parsed = JSON.parse(userInfo);
-        if (parsed.authToken) {
-          config.headers.Authorization = `Bearer ${parsed.authToken}`;
-        }
-      } catch (error) {
-        console.error('Failed to parse user info:', error);
-      }
-    }
-    return config;
-  },
-  (error) => {
-    return Promise.reject(error);
-  }
-);
+interface LoginResponse {
+  authToken: string;
+  userName: string;
+}
 
-api.interceptors.response.use(
-  (response) => response,
-  (error) => {
-    if (error.response?.status === 401) {
-      sessionStorage.removeItem('userInfo');
-      toast.error('세션이 만료되었습니다. 다시 로그인해주세요.');
+export const login = async (
+  credentials: LoginRequest
+): Promise<LoginResponse> => {
+  try {
+    const response = await apiClient.post<LoginResponse>(
+      '/auth/login',
+      credentials
+    );
 
-      if (window.location.pathname !== '/login') {
-        window.location.href = '/login';
-      }
+    const userInfo = {
+      authToken: response.data.authToken,
+      userName: response.data.userName,
+    };
+    sessionStorage.setItem('userInfo', JSON.stringify(userInfo));
+
+    return response.data;
+  } catch (error: any) {
+    if (error?.response?.status === 401) {
+      sessionStorage.removeItem('userInfo');
     }
-    return Promise.reject(error);
+    throw error;
   }
-);
-
-export default api;
+};
```

Comment:
```tsx
const userInfoStorage = createStorage('userInfo', window.sessionStorage);
```

이렇게 사용할 수 있도록 createStorage 함수를 하나 만들고

```suggestion
export const login = async (
  credentials: LoginRequest
): Promise<LoginResponse> => {
  try {
    const response = await apiClient.post<LoginResponse>(
      '/auth/login',
      credentials
    );

    userInfoStorage.set(response.data)

    return response.data;
  } catch (error: any) {
    if (error?.response?.status === 401) {
      userInfoStorage.clear()
    }
    throw error;
  }
};
```

이렇게 쓰일 수 있도록 해주면 어떨까요!?

Reply:
그리고 이렇게 쓰일 수 있도록 고차함수도 하나 만들어서 쓰면 좋답니다 ㅎㅎ

```suggestion
export const login = withError((credentials: LoginRequest) => {
    const response = await apiClient.post<LoginResponse>('/auth/login', credentials);
    return response.data;
}, {
  onScucess: userInfoStorage.set
  onError: (error) => {
    if (error?.response?.status === 401) {
      userInfoStorage.clear()
    }
  }
}
```

혹은 아예 storage 다루는 부분을 API 바깥쪽으로 위임할수도 있겠죠?

```js
const loginWithStorage = (...args) => {
  return login(...args)
    .then(result => (userInfoStorage.set(result), result))
    .catch(() => userInfoStorage.clear())
}
```

요로코롬...